test/e2e/performanceprofile/functests/13_llc/llc.go

```diff
@@ -0,0 +1,206 @@
+package __llc
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/ptr"
+
+	igntypes "github.com/coreos/ignition/v2/config/v3_4/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	machineconfigv1 "github.com/openshift/api/machineconfiguration/v1"
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	profilecomponent "github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components/profile"
+	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+
+	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/infrastructure"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/label"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/mcps"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/nodes"
+
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/poolname"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profilesupdate"
+)
+
+const (
+	llcEnableFileName            = "/etc/kubernetes/openshift-llc-alignment"
+	defaultIgnitionContentSource = "data:text/plain;charset=utf-8;base64"
+	defaultIgnitionVersion       = "3.2.0"
+	fileMode                     = 0420
+)
+
+var _ = Describe("[Performance] Last Level Cache", Label(string(label.OpenShift)), func() {
+	var (
+		workerRTNodes      []corev1.Node
+		perfProfile        *performancev2.PerformanceProfile
+		performanceMCP     string
+		isAMD              bool
+		ctx                context.Context = context.Background()
+		err                error
+		profileAnnotations = make(map[string]string)
+		poolName           string
+	)
+	testutils.CustomBeforeAll(func() {
+
+		workerRTNodes, err = nodes.GetByLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+		Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("error looking for the optional selector: %v", err))
+
+		isAMD, err = infrastructure.IsAMD(ctx, &workerRTNodes[0])
+		Expect(err).ToNot(HaveOccurred(), "Unable to fetch Vendor ID")
+		if !isAMD {
+			Skip("AMD LLC Tests can only run on AMD Hardware")
+		}
```

Comment:
I know that our focus is to run LLC tests on AMD but we could run these tests on any platform but I don't think it makes sense to skip here as the performance profile should configure the kubelet config and enable this feature on any platform.

Reply:
Right, for configuration tests, we can verify if the cpumanager policy align-cpus-by-uncorecache is enabled , but for tests where we want to check cpus are aligned as per the L3 cache id, can those tests be run only on specific Hardware and get skipped ?

Reply:
the best would be to check if the hardware has split L3 cache (aka split Uncore cache aka split LLC cache). Let's see how much code we need to check this

Reply:
In the latest commit i have removed this check since checking if the align-cpus-by-uncorecache cpumanager policy works , i.e kubelet starts successfully  doesn't require hardware.